apollo-router/src/axum_factory/compression/mod.rs

```diff
@@ -123,19 +124,25 @@ where {
                 }
             }
 
-            let buf = BytesMut::zeroed(64);
-            let mut partial_output = PartialBuffer::new(buf);
+            loop {
+                let buf = BytesMut::zeroed(1024);
+                let mut partial_output = PartialBuffer::new(buf);
 
-            match self.finish(&mut partial_output) {
-                Err(e) => {
-                    let _ = tx.send(Err(e.into())).await;
-                }
-                Ok(_) => {
-                    let len = partial_output.written().len();
-
-                    let mut buf = partial_output.into_inner();
-                    buf.resize(len, 0);
-                    let _ = tx.send(Ok(buf.freeze())).await;
+                match self.finish(&mut partial_output) {
```

Comment:
I looked at [the docs](https://docs.rs/flate2/1.0.26/flate2/write/struct.ZlibDecoder.html#method.finish) for `flate2`. The note suggests that we maybe shouldn't be using finish where the underlying stream of data is async.

Does that comment apply to us?

Reply:
here we know that there will be no more data after that because the stream has closed, so we can call finish